python_ta/debug/snapshot_tracer.py

```diff
@@ -39,37 +53,119 @@ def __init__(self, output_directory: Optional[str] = None, **kwargs) -> None:
             )
-            self._snapshot_counts += 1
+
+            line_number = frame.f_lineno - self._first_line + 1
+            self._add_svg_to_map(filename, line_number)
+
+    def _add_svg_to_map(self, filename: str, line: int) -> None:
+        """Add the SVG in filename to self._snapshots"""
+        with open(filename) as svg_file:
+            svg_content = svg_file.read()
+            self._snapshots.append(
+                {
+                    "lineNumber": line,
+                    "svg": svg_content,
+                }
+            )
 
     def __enter__(self):
         """Set up the trace function to take snapshots at each line of code."""
-        func_frame = inspect.getouterframes(inspect.currentframe())[1].frame
-        func_frame.f_trace = self._trace_func
+        self._func_frame = inspect.getouterframes(inspect.currentframe())[1].frame
+        self._func_frame.f_trace = self._trace_func
         sys.settrace(lambda *_args: None)
         return self
 
     def __exit__(self, exc_type, exc_val, exc_tb) -> None:
-        """Remove the trace function."""
+        """Remove the trace function and run any post-tracing functions."""
         sys.settrace(None)
-        inspect.getouterframes(inspect.currentframe())[1].frame.f_trace = None
+        self._func_frame.f_trace = None
+        if self.open_webstepper:
+            self._build_result_html()
+            self._open_html()
+
+    def _build_result_html(self) -> None:
+        """Build and write the Webstepper html to the output directory"""
+        current_dir = os.path.dirname(os.path.abspath(__file__))
+
+        html_content = self._read_original_html(current_dir)
+        soup = BeautifulSoup(html_content, "html.parser")
+
+        self._modify_bundle_import_path(current_dir, soup)
+        self._insert_data(soup)
+        self._write_generated_html(soup)
+
+    def _open_html(self) -> None:
+        """Open the generated HTML file in a web browser."""
+        index_html = f"file://{os.path.join(self.output_directory, 'index.html')}"
+        webbrowser.open(index_html, new=2)
+
+    def _read_original_html(self, current_dir: str) -> None:
+        """Read the original Webstepper html"""
+        original_index_html = os.path.join(current_dir, "webstepper", "index.html")
+        with open(original_index_html, "r") as file:
+            html_content = file.read()
+        return html_content
+
+    def _modify_bundle_import_path(self, current_dir: str, soup: BeautifulSoup) -> None:
+        """Modify the bundle path to the absolute path to the bundle"""
+        original_js_bundle = os.path.join(current_dir, "webstepper", "index.bundle.js")
+        soup.select("script")[0]["src"] = f"file://{original_js_bundle}"
+
+    def _insert_data(self, soup: BeautifulSoup) -> None:
+        """Insert the SVG array and code string into the Webstepper index HTML."""
+        code_script = f"<script>window.codeText=`{self._get_code()}` </script>\n"
```

Comment:
Make this a single `script` tag (with multiple statements) rather than two script tags.